pkg/envoy/ads/response.go

```diff
@@ -42,45 +43,50 @@ func (s *Server) sendTypeResponse(tURI envoy.TypeURI,
 	return nil
 }
 
-func (s *Server) sendAllResponses(proxy *envoy.Proxy, server *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer, cfg configurator.Configurator) {
-	log.Trace().Msgf("A change announcement triggered *DS update for proxy with SerialNumber=%s on Pod with UID=%s", proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
-
-	// Tracks the success of this full update of all its XDS paths. If a single XDS response path fails for this full update,
-	// the full updated will be considered as failed for metric purposes (success = false)
+// sendResponse takes a set of TypeURIs which will be called to generate the xDS resources
+// for, and will have them sent to the proxy server.
+// If no DiscoveryRequest is passed, an empty one for the TypeURI is created
+func (s *Server) sendResponse(typeURIsToSend mapset.Set,
+	proxy *envoy.Proxy,
+	server *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer,
+	request *xds_discovery.DiscoveryRequest,
+	cfg configurator.Configurator) error {
 	success := true
-	defer xdsPathTimeTrack(time.Now(), log.Info(), ADSUpdateStr, proxy.GetCertificateSerialNumber().String(), &success)
+	if typeURIsToSend.Cardinality() == len(envoy.XDSResponseOrder) {
```

Comment:
What is the purpose of the comparison of the size of the set and the list of responses?
Why only track the time it took when the sizes are equal?
A comment would be a good step forward.

Reply:
Since now the function takes arbitrary xDS parts to make and send, this is the best way I could find to identify a "full update". This is in order to account for how much time does a full ADS update take.

For each individual xDS path, time tracking happens [here](https://github.com/openservicemesh/osm/pull/2844/files#diff-7a1f450fbabb0088828083ed90ad3ccc49ed4edd9ae6f58ad5bdcdd158b2dda8R27)

I can, of course, drop a comment.